### PR TITLE
[Documentation] Adding API documentation for CloudSync

### PR DESCRIPTION
**Pull request by nitu-s-gupta**
Signed-off-by: Nitu Gupta <nitu.gupta@samsung.com>

# Description
Added the API document for CloudSync

Fixes # (#598)

## Type of change

Please delete options that are not relevant.
- [X] Documentation update

# How Has This Been Tested?
You can see the results in https://editor.swagger.io/

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: api/cloudsync_api.yaml [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    SubscribeResponse:
+      properties:
+        Message:
+          type: string
+          example: Successfully subscribed to the topic homeedge/livingroom 
+    PublishResponse:
+      properties:
+        Message:
+          type: string
+          example: Data published successfully to Cloudtcp://broker.hivemq.com:1883
```
Comment (t25kim) [inline api/cloudsync_api.yaml:126]: Please use `publishbody url` instead of hivemq address.

File: api/cloudsync_api.yaml [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+          type: string
+          example: doggie
+        url:
+          type: string
+          example: ec2-184-73-83-74.compute-1.amazonaws.com
+    SubscribeResponse:
+      properties:
+        Message:
+          type: string
+          example: Successfully subscribed to the topic homeedge/livingroom
```
Comment (t25kim) [inline api/cloudsync_api.yaml:121]: Please use `subscribebody topic` instead of `homeedge/livingroom`.

File: api/cloudsync_api.yaml (hunk 1 above)
Comment (nitu-s-gupta) [reply api/cloudsync_api.yaml]: done!!

File: api/cloudsync_api.yaml (hunk 2 above)
Comment (nitu-s-gupta) [reply api/cloudsync_api.yaml]: done!!!

**Review by t25kim — APPROVED** [review]
LGTM Thanks!

**Review by suresh-lc — APPROVED** [review]
+1

**Review by MoonkiHong — APPROVED** [review]
+1

**Review by tdrozdovsky — APPROVED** [review]
LGTM

**Pull request merged**